lib/Geo/GeoUtils.cpp

```diff
@@ -99,20 +101,20 @@ void GeoUtils::scanIntervals(QueryParams const& params, std::vector<S2CellId> co
     // not need
     // to look at [47|1|40] or [47|11|60] because these cells don't intersect,
     // but polygons indexed with exact cell id [47|11] still might.
-    std::set<S2CellId> parentSet;
+    arangodb::HashSet<uint64_t> parentSet;
```

Comment:
I am unsure if this change is correct, because we are switching from a sorted set to an unsorted set here.

And a few lines further down it seems like the sorting of this set is relevant, because:
```
    for (uint64_t exact : parentSet) {
      sortedIntervals.emplace_back(S2CellId(exact), S2CellId(exact));
    }
```
Stuff is inserted in set-order into a vector named `sortedIntervals`. My assumption is that this vector needs to be sorted, due to the variable name.

Reply:
just below that:
`std::sort(sortedIntervals.begin(), sortedIntervals.end(), Interval::compare);`

Reply:
Ok, didn't see this, as the sort was there before already and not part of the diff.